src/core/ashp.jl

```diff
@@ -0,0 +1,127 @@
+# REopt®, Copyright (c) Alliance for Sustainable Energy, LLC. See also https://github.com/NREL/REopt.jl/blob/master/LICENSE.
+
+struct ASHP <: AbstractThermalTech
+    min_kw::Real
+    max_kw::Real
+    installed_cost_per_kw::Real
+    om_cost_per_kw::Real
+    macrs_option_years::Int
+    macrs_bonus_fraction::Real
+    can_supply_steam_turbine::Bool
+    cop_heating::Array{Float64,1}
+    cop_cooling::Array{Float64,1}
+    can_serve_dhw::Bool
+    can_serve_space_heating::Bool
+    can_serve_process_heat::Bool
+    can_serve_cooling::Bool
+end
+
+
+"""
+ASHP
+
+If a user provides the `ASHP` key then the optimal scenario has the option to purchase 
+this new `ASHP` to meet the heating load in addition to using the `ExistingBoiler`
+to meet the heating load. 
+
+```julia
+function ASHP(;
+    min_mmbtu_per_hour::Real = 0.0, # Minimum thermal power size
+    max_mmbtu_per_hour::Real = BIG_NUMBER, # Maximum thermal power size
+    installed_cost_per_mmbtu_per_hour::Union{Real, nothing} = nothing, # Thermal power-based cost
+    om_cost_per_mmbtu_per_hour::Union{Real, nothing} = nothing, # Thermal power-based fixed O&M cost
+    macrs_option_years::Int = 0, # MACRS schedule for financial analysis. Set to zero to disable
+    macrs_bonus_fraction::Real = 0.0, # Fraction of upfront project costs to depreciate under MACRS
+    can_supply_steam_turbine::Union{Bool, nothing} = nothing # If the boiler can supply steam to the steam turbine for electric production
+    cop_heating::Array{<:Real,2}, # COP of the heating (i.e., thermal produced / electricity consumed)
+    cop_cooling::Array{<:Real,2}, # COP of the cooling (i.e., thermal produced / electricity consumed)
+    can_serve_dhw::Union{Bool, Nothing} = nothing # If ASHP can supply heat to the domestic hot water load
+    can_serve_space_heating::Union{Bool, Nothing} = nothing # If ASHP can supply heat to the space heating load
+    can_serve_process_heat::Union{Bool, Nothing} = nothing # If ASHP can supply heat to the process heating load
+    can_serve_cooling::Union{Bool, Nothing} = nothing # If ASHP can supply heat to the cooling load
+)
+```
+"""
+function ASHP(;
+        min_mmbtu_per_hour::Real = 0.0,
+        max_mmbtu_per_hour::Real = BIG_NUMBER,
```

Comment:
We should use `ton` instead of `mmbtu_per_hour` for heat pump thermal production. That's how they typically state the capacity in the industry. So this should apply to min, max, installed cost, O&M cost, and any incentives which I see we don't have a generic $/ton or percent-based incentive currently.

Reply:
I converted mmbtu to ton for ASHP capacity here https://github.com/NREL/REopt.jl/commit/032e0d6cd659b47851b2a41c45bfe06c434afd4f

Reply:
The unit "ton" is actually a thermal **power/rate**, so no need to add "per_hour" for the min/max/installed_cost/om_cost fields.

Reply:
addressed here: https://github.com/NREL/REopt.jl/pull/372/commits/03f7172af860c06d6dbe994f343afb5d299d0572

and here: https://github.com/NREL/REopt.jl/pull/372/commits/012cc400bdbf7e3c75f460810db26c4e02e9e6c3